Orange/widgets/utils/itemmodels.py

```diff
@@ -222,39 +239,34 @@ def __setitem__(self, i, value):
     def wrap(self, table):
         self.beginResetModel()
         self._table = table
+        rows, cols = range(len(table)), range(len(table[0]) if table else 0)
+        self._roleData = [[{} for _ in cols] for _ in rows]
```

Comment:
PyTableModel was, likewise, meant for smaller tables like confusion matrices, Test & Score results etc. But given how useful it appears to be, I'll take you up on it and make it even better.